src/framework/components/anim/component.js

```diff
@@ -0,0 +1,441 @@
+        /**
+         * @function
+         * @name pc.AnimComponentLayer#play
+         * @description Start playing the animation in the current state.
+         * @param {string} name - If provided, will begin playing from the start of the state with this name.
+         */
+        play: function (name) {
+            this._controller.play(name);
+        },
+
+        /**
+         * @function
+         * @name pc.AnimComponentLayer#pause
+         * @description Start playing the animation in the current state.
+         */
+        pause: function () {
+            this._controller.pause();
+        },
+
+        /**
+         * @function
+         * @name pc.AnimComponentLayer#reset
+         * @description Reset the animation component to it's initial state, including all parameters. The system will be paused.
+         */
+        reset: function () {
+            this._controller.reset();
+        },
+
+        update: function (dt) {
+            this._controller.update(dt);
+        },
+
+        /**
+         * @function
+         * @name pc.AnimComponentLayer#assignAnimation
+         * @description Associates an animation with a state in the loaded state graph. If all states are linked and the pc.AnimComponent.activate value was set to true then the component will begin playing.
+         * @param {string} stateName - The name of the state that this animation should be associated with.
+         * @param {object} animTrack - The animation track that will be assigned to this state and played whenever this state is active.
+         */
+        assignAnimation: function (stateName, animTrack) {
+            this._controller.assignAnimation(stateName, animTrack);
+
+            if (this._component.activate) {
+                for (var i = 0; i < this._component.data.layers.length; i++) {
+                    if (!this._component.data.layers[i].playable) {
+                        return;
+                    }
+                    this._component.playing = true;
+                }
+            }
+        },
+
+        /**
+         * @function
+         * @name pc.AnimComponentLayer#removeStateAnimations
+         * @description Removes animations from a state in the loaded state graph.
+         * @param {string} stateName - The name of the state that should have its animation tracks removed.
+         */
+        removeStateAnimations: function (stateName) {
+            this._controller.removeStateAnimations(stateName);
+        },
+
+        getParameterValue: function (name, type) {
+            this._controller.getParameterValue(name, type);
+        },
+
+        setParameterValue: function (name, type, value) {
+            this._controller.setParameterValue(name, type, value);
+        }
+    });
+
+    Object.defineProperties(AnimComponentLayer.prototype, {
+        /**
+         * @name pc.AnimComponentLayer#name
+         * @property {string} name - Returns the name of the layer
+         */
+        name: {

```

Comment:
The jsdoc for all properties in this block define the properties as objects (since `@property` is used), resulting in incorrect typings as such:

```typescript
class AnimComponentLayer {
    ...
    name: { name: string }; // Should be `name: string;`
    ...
}
```

I believe using `@type` instead of `@property` would fix the issue:
```
/**
 * @readonly
 * @name pc.AnimComponentLayer#name
 * @type {string}
 * @description Returns the name of the layer
 */
```

Reply:
Updated this too